qubes-rpc/qubes.StartApp

```diff
@@ -1,25 +1,50 @@
-#!/bin/sh
+#!/usr/bin/python3 --
 
-if [ -z "$1" ]; then
-    echo "This service requires an argument" >&2
-    exit 1
-fi
+import sys, os, pwd
+import qubesdb
+from qubesagent.xdg import launch
 
-app_basename="$1.desktop"
+def main(argv):
+    try:
+        (arg0, arg,) = argv
+    except ValueError:
+        print('This service requires an argument', file=sys.stderr)
+        sys.exit(1)
+    app_name = os.path.join('applications', arg + '.desktop')
+    try:
+        xdg_data_home = os.environ['XDG_DATA_HOME']
+    except KeyError:
+        xdg_data_home = os.path.join(pwd.getpwuid(os.getuid()).pw_dir, '.local/share')
```

Comment:
I believe canonical way for doing this is `os.path.expanduser('~/.local/share')`, possibly with a check if `$HOME` is set.